stac_generator/vector_polygon/generator.py

```diff
@@ -0,0 +1,73 @@
+import fiona
+import pandas as pd
+import pystac
+from pystac.extensions.projection import ItemProjectionExtension
+from shapely.geometry import mapping
+
+from stac_generator.base.generator import StacGenerator
+from stac_generator.base.schema import SourceConfig, StacCatalogConfig, StacCollectionConfig
+
+
+class VectorPolygonGenerator(StacGenerator[SourceConfig]):
+    def __init__(
+        self,
+        source_df: pd.DataFrame,
+        collection_cfg: StacCollectionConfig,
+        catalog_cfg: StacCatalogConfig | None = None,
+        href: str | None = None,
+    ) -> None:
+        super().__init__(
+            source_df=source_df,
+            collection_cfg=collection_cfg,
+            catalog_cfg=catalog_cfg,
+            href=href,
+        )
+
+    def create_item_from_config(self, source_cfg: SourceConfig) -> list[pystac.Item]:
+        if source_cfg.location.endswith(".zip"):  # Assume zip shape archive
+            shapefile_name = source_cfg.location.split("/")[-1].replace(
+                ".zip", ".shp"
+            )  # Extract .shp file name
+
+            if source_cfg.location.startswith("http"):  # Remote file (HTTP/HTTPS)
```

Comment:
Im just curious why you used this vsicurl method rather than the one I suggested on the last PR. I have been trying to test how it works locally but as you have seen I am having computer problems.

Reply:
I can't get it to work this way so lets figure it out tomorrow.